libbeat/publisher/processing/default.go

```diff
@@ -143,10 +144,17 @@ var WithHost modifier = builtinModifier(func(info beat.Info) mapstr.M {
 // pipeline.
 func WithAgentMeta() modifier {
 	return builtinModifier(func(info beat.Info) mapstr.M {
+		var hostname string
+		if features.FQDN() {
```

Comment:
small nit: this check is on different places, maybe we could provide a helper in info accepting a flag and returning proper hostname

Reply:
Good idea, done.